File: src/org/usfirst/frc5293/subsystems/CameraRingLight.java
package org.usfirst.frc5293.subsystems;

import edu.wpi.first.wpilibj.Relay;
import edu.wpi.first.wpilibj.command.Subsystem;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
import org.usfirst.frc5293.Devices;

public class CameraRingLight extends Subsystem {

    @Override
    protected void initDefaultCommand() {
        // input will set callback
    }

    public void enable() {
        set(true);
    }

    public void disable() {
        set(false);
    }

    public void set(boolean isEnabled) {
        final Relay.Value value =
                (isEnabled)
                ? Relay.Value.kForward
                : Relay.Value.kOff;

        getRelay().set(value);
        SmartDashboard.putBoolean("Camera Ring Light Enabled?", isEnabled);
    }

    public boolean get() {
        return getRelay().get().equals(Relay.Value.kOn);
    }

    public void swap() {
        set(!get());
    }

    private Relay getRelay() {
        return Devices.getCameraRingLight().getRelay();
    }
}

File: src/org/usfirst/frc5293/subsystems/CameraRingLight.kt
package org.usfirst.frc5293.subsystems

import edu.wpi.first.wpilibj.Relay
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
import org.usfirst.frc5293.Devices
import org.usfirst.frc5293.subsystems.util.EmptySubsytem

class CameraRingLight : EmptySubsytem() {

    private val relay by lazy { Devices.cameraRingLight.relay }

    var isEnabled: Boolean
        get() = relay.get() == Relay.Value.kOn
        set(isEnabled) {
            val value =
                if (isEnabled)
                    Relay.Value.kForward
                else
                    Relay.Value.kOff

            relay.set(value)
            SmartDashboard.putBoolean("Camera Ring Light Enabled?", isEnabled)
        }

    fun swap() {
        isEnabled = !isEnabled
    }
}
